consensus/misc/eip4844.go

```diff
@@ -45,16 +47,20 @@ func CalcExcessDataGas(parentExcessDataGas *big.Int, newBlobs int) *big.Int {
 
 // FakeExponential approximates factor * e ** (num / denom) using a taylor expansion
 // as described in the EIP-4844 spec.
-func FakeExponential(factor, num, denom *big.Int) *big.Int {
-	output := new(big.Int)
-	numAccum := new(big.Int).Mul(factor, denom)
-	for i := 1; numAccum.Sign() > 0; i++ {
-		output.Add(output, numAccum)
-		numAccum.Mul(numAccum, num)
-		iBig := big.NewInt(int64(i))
-		numAccum.Div(numAccum, iBig.Mul(iBig, denom))
+func FakeExponential(factor, denom *uint256.Int, edg *big.Int) (*uint256.Int, error) {
+	numerator, overflow := uint256.FromBig(edg)
+	if overflow {
+		return nil, fmt.Errorf("FakeExponential: overflow converting excessDataGas: %v", edg)
+	}
+	output := uint256.NewInt(0)
+	numeratorAccum := new(uint256.Int).Mul(factor, denom)
+	for i := 1; numeratorAccum.Sign() > 0; i++ {
+		output.Add(output, numeratorAccum)
+		numeratorAccum.Mul(numeratorAccum, numerator)
```

Comment:
Once you are using uint256 you should be using MulOverflow to check for overflow. 

Ultimately we want to change edg to uint256 as well I think since Erigon seems to avoid using big.Int in general, but this is a step in that direction at least.

Reply:
Yes, I was thinking about changing edg to uint256. Will do it